src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -133,11 +140,107 @@ Copyright (c) .NET Foundation. All rights reserved.
     <TargetLatestRuntimePatch Condition="'$(SelfContained)' != 'true' ">false</TargetLatestRuntimePatch>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(RuntimeFrameworkVersion)' == ''">
-    <RuntimeFrameworkVersion Condition="'$(TargetLatestRuntimePatch)' == 'true' ">$(LatestNetCorePatchVersion)</RuntimeFrameworkVersion>
-    <RuntimeFrameworkVersion Condition="'$(TargetLatestRuntimePatch)' != 'true' ">$(DefaultNetCorePatchVersion)</RuntimeFrameworkVersion>
+  <!-- Determine the default values of TargetLatestAspNetCoreRuntimePatch -->
+  <PropertyGroup Condition="'$(TargetLatestAspNetCoreRuntimePatch)' == ''">
+    <TargetLatestAspNetCoreRuntimePatch>$(TargetLatestRuntimePatch)</TargetLatestAspNetCoreRuntimePatch>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(RuntimeFrameworkName)' == ''">
+    <RuntimeFrameworkName>Microsoft.NETCore.App</RuntimeFrameworkName>
+  </PropertyGroup>
+
+  <!--
+    Determine the value of RuntimeFrameworkVersion.
+
+    It is expected that the Microsoft.AspNetCore.App/All and Microsoft.NETCore.App will always ship together with the same version numbers for 2.1 and 2.2.
+    Therefore, this uses the same property fro both implicit package references.
+  -->
+  <Choose>
+    <When Condition=" '$(RuntimeFrameworkName)' == 'Microsoft.AspNetCore.All' " >
```

Comment:
I understood the rationale for why we can have just one property is that the versions will align the versions of these shared frameworks. It makes me wonder why we have separate logic for setting the version of each of them here.

Reply:
Presumably because when we have the core shared framework, the baseline is still 2.1.0, but we've already set the precedent of a different baseline of 2.1.1 for ASP.NET?

Reply:
Yup, that's why. I only added the separate properties because aspnetcore's baseline is 2.1.1 (for now.)